sequencer/src/genesis.rs

```diff
@@ -487,11 +533,17 @@ mod test {
         }
         .to_string();
 
-        let genesis: Genesis = toml::from_str(&toml).unwrap_or_else(|err| panic!("{err:#}"));
-        let rpc_url = "https://ethereum-sepolia.publicnode.com";
+        let mut genesis: Genesis = toml::from_str(&toml).unwrap_or_else(|err| panic!("{err:#}"));
+
+        setup_test();
+
+        let anvil = Anvil::new().spawn();
+        let (_wallet, contract) = deploy_fee_contract_for_test(&anvil).await?;
+
+        genesis.chain_config.fee_contract = Some(contract.address());
```

Comment:
can we just pass the variable in the toml! macro instead of overwriting it. So, we deploy the contracts first -> pass the address everywhere in toml! {} -> parse the toml -> validate

Reply:
addressed here https://github.com/EspressoSystems/espresso-sequencer/commit/5f8cbcfbda4e978a2f5313e2fb217ed4ef06fbda